components/dashboard/src/workspaces/VideoCarousel.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import React, { useState } from "react";
+import { trackVideoClick } from "../Analytics";
+
+import "lite-youtube-embed/src/lite-yt-embed.css";
+import "lite-youtube-embed/src/lite-yt-embed";
+
+interface Video {
+    id: string;
+    title: string;
+    analyticsLabel: string;
+}
+
+const videos: Video[] = [
+    { id: "1ZBN-b2cIB8", title: "Gitpod in 120 seconds", analyticsLabel: "gitpod-demo" },
+    { id: "zhZNnzFlZnY", title: "Getting started with Gitpod", analyticsLabel: "getting-started-with-gitpod" },
+    { id: "kuoHM2bpBqY", title: "Fully automate your dev setup", analyticsLabel: "automate-gitpod-setup" },
+    { id: "_CwFzCbAsoU", title: "Personalise your workspace", analyticsLabel: "personalise-gitpod-workspace" },
+];
+
+declare global {
+    namespace JSX {
+        interface IntrinsicElements {
+            "lite-youtube": any;
+        }
+    }
+}
+
+export const VideoCarousel: React.FC = () => {
+    const [currentVideo, setCurrentVideo] = useState(0);
+
+    const handleDotClick = (index: number) => {
+        setCurrentVideo(index);
+    };
+
+    const onPlayerStateChange = (index: number) => {
+        trackVideoClick(videos[index].analyticsLabel);
+    };
+
+    return (
+        <div className="video-carousel">
+            <div className="video-container">
+                {videos.map((video, index) => (
+                    <div key={video.id} style={{ display: index === currentVideo ? "block" : "none" }}>
+                        <lite-youtube
+                            key={videos[currentVideo].id}
```

Comment:
Because we're mapping over videos here and using the global state `currentVideo`, we will end up with many `<lite-youtube>` with the same key. We should ensure those are [unique](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key) and because the parent already has a unique key, we should instead just either just our `Tabs` component, or dynamically render the currently selected video and keep the rest unmnounted.

Reply:
```tsx
key = {video.id}
```

was leading to that bug 🐛, in which it kept getting played if we even changed the video (clicked on dot 2)

Reply:
I understand the bug happened while the key was set to the actual video id, but we are very much doing two things react does not expect us to:
1. Use non-static `key` props
2. Have multiple `key` props have the same values

As mentioned, if we want to not have the videos continue playing, we should ditch the `.map` over `videos` and only render `videos[currentVideo]`.

Reply:
Resolved in [`12ac117`](https://github.com/gitpod-io/gitpod/pull/20021/commits/12ac117ae2b9e3789e1fd251ef3407ca21d85ece)